### ISSUE TITLE
field.file with useFalRelation nested in form.section is not saved into sys_file_reference

### ISSUE THREAD

**Issue by iamandrewluca** [FluidTYPO3/flux#1493]
Environment
```
TYPO3: 7.6.23
flux: 8.2.1
fluidcontent: 5.2.0
PHP: 7
```
Configuration

this 3 fields are saved to `sys_file_reference`
```xml
<flux:form.sheet name="hero">
    <flux:field.inline.fal name="supportedImage" collapseAll="true" minItems="1"/>
    <flux:field.inline.fal name="heroBackgroundImage" collapseAll="true" minItems="1"/>
    <flux:field.inline.fal name="videoImage" collapseAll="true" minItems="1"/>
</flux:form.sheet>
```

this field with `useFalRelation` is not saved to `sys_file_reference` or it is saved in another place?
```xml
<flux:form.sheet name="timeLine">
    <flux:form.section name="products" label="Products">
        <flux:form.object name="product" label="Product">
            <flux:field.file name="image" useFalRelation="1" minItems="1" showThumbnails="1" allowed="jpg,png"/>
        </flux:form.object>
    </flux:form.section>
</flux:form.sheet>
```

Column `pi_flexform` content
```xml
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<T3FlexForms>
    <data>
        <sheet index="hero">
            <language index="lDEF">
                <field index="supportedImage">
                    <value index="vDEF">1</value>
                </field>
                <field index="heroBackgroundImage">
                    <value index="vDEF">1</value>
                </field>
                <field index="videoImage">
                    <value index="vDEF">1</value>
                </field>
            </language>
        </sheet>
        <sheet index="timeLine">
            <language index="lDEF">
                <field index="products">
                    <el index="el">
                        <section index="1">
                            <itemType index="product">
                                <el>
                                    <field index="image">
                                        <value index="vDEF">446</value>
                                    </field>
                                    <field index="id">
                                        <value index="vDEF">a135aae117d3026c199e69e9a3226ac5eb2b7a88</value>
                                    </field>
                                </el>
                            </itemType>
                            <itemType index="_TOGGLE">0</itemType>
                        </section>
                    </el>
                </field>
            </language>
        </sheet>
    </data>
</T3FlexForms>
```

**Comment by NamelessCoder** [comment]
I believe this is entirely on purpose and done that way because TYPO3 itself does not support writing references *unless you specifically create an IRRE to sys_file_reference*, and doing that comes with a bunch of side effects such as problems maintaining the reference index and possible duplicate relations if you use more than one such field in your template.

The explanation lies in the TYPO3 internals and how it handles IRRE references. In short, a table and field name is used to identify where the reference exists - but the field name has no support for referencing a specific FlexForm field; it can only reference the field name. And that's the key: if your field name is the same for all objects, it fails. If the field name is different, your FE output would fail or be very complicated to produce.

TL;DR: TYPO3 doesn't support FAL IRRE in FlexForm sections, so Flux does not support it either.

**Comment by cweiske** [comment]
See #759.
